### PR TITLE
Small optimizations (#74)

### PR DESCRIPTION
**Pull request by SergeiPavlov**
* Optimize VisitMethodCall()

* Optimize Provider constructor

* Use Array.Empty<T>() instead of allocating Empty array

* Get rid of unused PriorityQueue<> class

* VisitNullable()

### CODE REVIEW

**Comment by alex-kulakov** [comment]
> Get rid of unused PriorityQueue<> class

No public classes deletions allowed, make it [Obsolete]

File: Orm/Xtensive.Orm/Orm/Rse/Providers/Provider.cs [hunk 1]
```diff
@@ -136,7 +127,7 @@ protected virtual string ParametersToString()
     /// </summary>
     /// <param name="type">The type of the provider.</param>
     /// <param name="sources"><see cref="Sources"/> property value.</param>
-    protected Provider(ProviderType type, params Provider[] sources)
+    protected Provider(ProviderType type, IReadOnlyList<Provider> sources)
```
Comment (alex-kulakov) [inline Orm/Xtensive.Orm/Orm/Rse/Providers/Provider.cs:130]: Can you explain what was the problem here? Native array is better, isn't it?
Was it for safety reasons so user code wouldn't be able to change items?

This is a breaking change which generates a breaking change of the ```ExecutableProvider``` ctor, so I need to evaluate its pros.

**Review by alex-kulakov — CHANGES_REQUESTED** [review]

File: Orm/Xtensive.Orm/Orm/Providers/SqlSelectProcessor.cs [hunk 2]
```diff
@@ -562,6 +530,13 @@ public void Visit(SqlExpression sqlExpression)
       sqlExpression.AcceptVisitor(this);
     }
 
+    public void VisitNullable(SqlExpression sqlExpression)
```
[the remark this answers is not in the document]
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/Providers/SqlSelectProcessor.cs]: done

File: Orm/Xtensive.Orm/Orm/Rse/Providers/Provider.cs (hunk 1 above)
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/Rse/Providers/Provider.cs:130]: The only reason: make `Provider.Sources` readonly collection

**Comment by SergeiPavlov** [comment]
> > Get rid of unused PriorityQueue<> class
> 
> No public classes deletions allowed, make it [Obsolete]. Removing the test for this collection is OK, though.

`PrioriyQueue` is standard class in .NET6+
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.priorityqueue-2?view=net-7.0

**Review by alex-kulakov — APPROVED** [review]

**Pull request merged**